am-lib/src/main/java/uk/gov/hmcts/reform/amlib/internal/aspects/ErrorHandlingAspect.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.hmcts.reform.amlib.internal.aspects;
+
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.jdbi.v3.core.JdbiException;
+import uk.gov.hmcts.reform.amlib.exceptions.PersistenceException;
+
+@Aspect
+public class ErrorHandlingAspect {
+
+    @Around("execution(public * uk.gov.hmcts.reform.amlib.*Service.*(..))")
+    public Object around(ProceedingJoinPoint joinPoint) throws Throwable {
+        try {
+            return joinPoint.proceed();
+        } catch (JdbiException ex) {
```

Comment:
is there cases where something other than a Jdbi exception would want us to throw our persistence exception?

Reply:
I do not think so. On one hand all low level SQL exceptions are wrapped in one of the JDBI exceptions that all inherit from `JdbiException`. On the other hand `PersistenceException` should be only used when dealing with database or other persistent data store of choice.